src/wasm-interpreter.h

```diff
@@ -2173,21 +2225,25 @@ template<typename GlobalManager, typename SubType> class ModuleInstanceBase {
     }
   }
 
-  template<class LS> Address getFinalAddress(LS* curr, Literal ptr) {
+  template<class LS>
+  Address getFinalAddress(LS* curr, Literal ptr, Index bytes) {
     Address memorySizeBytes = memorySize * Memory::kPageSize;
     uint64_t addr = ptr.type == Type::i32 ? ptr.geti32() : ptr.geti64();
     trapIfGt(curr->offset, memorySizeBytes, "offset > memory");
     trapIfGt(addr, memorySizeBytes - curr->offset, "final > memory");
     addr += curr->offset;
-    trapIfGt(curr->bytes, memorySizeBytes, "bytes > memory");
-    checkLoadAddress(addr, curr->bytes);
+    trapIfGt(bytes, memorySizeBytes, "bytes > memory");
```

Comment:
Does this only happen if the memory has size 0?

Reply:
I think you're right, given that a single page is 64K and the bytes field is <= 8 for now.

Can leave as a separate followup if we're sure that would be future-proof.

Reply:
Sounds good, I was just checking my understanding. No changes necessary, I think.